clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -41,7 +41,9 @@
 #include "clang/Basic/CharInfo.h"
 #include "clang/Basic/CodeGenOptions.h"
 #include "clang/Basic/Diagnostic.h"
+#include "clang/Basic/DiagnosticFrontend.h"
```

Comment:
This is a LOT of include additions.  Could we instead create a function "elsewhere" (new .cpp file perhaps?) that does all the work, perhaps just `CodeGenModule::DiagDemanglerFailures`?

Reply:
Sorry, most of those are a result of multiple iterations of this patch. 
The only necessary one is `llvm/Demangle/Demangle.h`.
Is that all right to add one include or should I still extract the code? 
Removing all the other includes for now.

Reply:
Ok,just the 1 is fine then.

Reply:
Thanks!